### PR TITLE
chore(*) remove Kubernetes submodule

### PR DESCRIPTION
**Pull request by jpeach**
### Summary

Remove the Kubernetes Go submodule.

### Full changelog

N/A

### Issues resolved

Updates #2200 

### Documentation

N/A

### Testing

- [x] Unit tests
- [x] E2E tests
- [ ] Manual testing on Universal
- [ ] Manual testing on Kubernetes 

### Backwards compatibility

- [x] Add `backport-to-stable` label if the code is backwards compatible. Otherwise, list breaking changes.

### CODE REVIEW

**Comment by lobkovilya** [comment]
I think a good amount of Makefile targets also could be deleted (probably all of them), as well as `Dockerfile` and `PROJECT`

**Comment by jpeach** [comment]
> I think a good amount of Makefile targets also could be deleted (probably all of them), as well as `Dockerfile` and `PROJECT`

I removed these and cleaned up the Makefile a bit/

File: pkg/plugins/resources/k8s/native/config/rbac/role.yaml [hunk 1]
```diff
@@ -66,14 +66,6 @@ rules:
...
   - update
   - watch
-- apiGroups:
-  - kuma.io
-  resources:
-  - proxytemplates/status
-  verbs:
-  - get
-  - patch
-  - update
```
Comment (jpeach) [inline pkg/plugins/resources/k8s/native/config/rbac/role.yaml:76]: ProxyTemplate doesn't have a Status field, and it doesn't make sense for it to have a status subresource. So this change is correct, but not sure how it got in there in the first place.

**Comment by lobkovilya** [comment]
@jpeach did you check if CI is just flaky or it's something connected to the changes?

**Comment by jpeach** [comment]
> @jpeach did you check if CI is just flaky or it's something connected to the changes?

Yeh some  sort of flake. I was able to re-run it to make it clear while I was failing to try to set it up and reproduce locally.

**Review by lobkovilya — APPROVED** [review]

**Pull request merged**